tests/runtime-tests/src/lib.rs

```diff
@@ -210,169 +197,11 @@ fn copy_manifest(
-            }
-            if let Some(status) = spin.try_wait()? {
-                anyhow::bail!(
-                    "Spin exited early with status code {:?}\n{}{}",
-                    status.code(),
-                    spin.stdout.output_as_str().unwrap_or("<non-utf8>"),
-                    spin.stderr.output_as_str().unwrap_or("<non-utf8>")
-                );
-            }
-
-            std::thread::sleep(std::time::Duration::from_millis(250));
-        }
-        anyhow::bail!(
-            "`spin up` did not start server or error after 20 seconds. stderr:\n\t{}",
-            spin.stderr.output_as_str().unwrap_or("<non-utf8>")
-        )
-    }
-
-    /// The port Spin is running on
-    fn port(&self) -> u16 {
-        self.port
-    }
-
-    fn try_wait(&mut self) -> std::io::Result<Option<std::process::ExitStatus>> {
-        self.process.try_wait()
-    }
-}
-
-impl Drop for Spin {
-    fn drop(&mut self) {
-        kill_process(&mut self.process);
-    }
-}
-
-fn kill_process(process: &mut std::process::Child) {
-    #[cfg(windows)]
-    {
-        let _ = process.kill();
-    }
-    #[cfg(not(windows))]
-    {
-        let pid = nix::unistd::Pid::from_raw(process.id() as i32);
-        let _ = nix::sys::signal::kill(pid, nix::sys::signal::SIGTERM);
-    }
-}
-
-/// Uses a track to ge a random unused port
-fn get_random_port() -> anyhow::Result<u16> {
-    Ok(std::net::TcpListener::bind("localhost:0")?
-        .local_addr()?
-        .port())
-}
-
-/// Helper for reading from a child process stream in a non-blocking way
-struct OutputStream {
-    rx: std::sync::mpsc::Receiver<Result<Vec<u8>, std::io::Error>>,
-    buffer: Vec<u8>,
-}
-
-impl OutputStream {
-    fn new<R: Read + Send + 'static>(mut stream: R) -> Self {
-        let (tx, rx) = std::sync::mpsc::channel();
-        std::thread::spawn(move || {
-            let mut buffer = vec![0; 1024];
-            loop {
-                let msg = stream.read(&mut buffer).map(|n| buffer[..n].to_vec());
-                if let Err(e) = tx.send(msg) {
-                    if let Err(e) = e.0 {
-                        eprintln!("Error reading from stream: {e}");
-                    }
-                    break;
-                }
-            }
-        });
-        Self {
-            rx,
-            buffer: Vec::new(),
-        }
-    }
-
-    /// Get the output of the stream so far
-    fn output(&mut self) -> &[u8] {
-        while let Ok(Ok(s)) = self.rx.try_recv() {
-            self.buffer.extend(s);
-        }
-        &self.buffer
-    }
-
-    /// Get the output of the stream so far
-    ///
-    /// Returns None if the output is not valid utf8
-    fn output_as_str(&mut self) -> Option<&str> {
-        std::str::from_utf8(self.output()).ok()
-    }
+pub enum TestResult {
+    /// The test passed
+    Pass,
+    /// Wasm errored (the wasm error, additional error info)
+    Err(String, String),
+    /// Wasm failed to run (additional error info)
+    Fatal(String),
```

Comment:
Bikeshedding, maybe something like `Err` -> `Fail` and `Fatal` -> `RuntimeError` might be more clear?